src/web/services/search.service.ts

```diff
@@ -162,14 +162,23 @@ export class SearchService {
 
     let masqueradeGoogleId: string = '';
     for (const instructor of instructors.instructors) {
-      const instructorPrivilege: InstructorPrivilege | undefined = instructorPrivileges.shift();
       if (instructor.googleId != null
-          && (instructorPrivilege != null && instructorPrivilege.privileges.courseLevel.canModifyInstructor
-              || instructor.role === InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_COOWNER)) {
+          && instructor.role === InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_COOWNER) {
         masqueradeGoogleId = instructor.googleId;
         break;
       }
     }
+    // there's no instructor being co-owner; there's always at least one instructor with "modify instructor" permission
+    if (masqueradeGoogleId === '') {
+      for (const instructor of instructors.instructors) {
+        const instructorPrivilege: InstructorPrivilege | undefined = instructorPrivileges.shift();
+        if (instructor.googleId != null
+            && (instructorPrivilege != null && instructorPrivilege.privileges.courseLevel.canModifyInstructor)) {
```

Comment:
nit: While it works here, in my experience it's usually a bad idea to use the equality operator in js unless you have to because it is extremely error prone and a source of a lot of bugs in js/ts code. In fact, in both cases we actually want to check for undefined and not null here.

In most cases, unless we specifically want to check values of different types, it's usually preferred to:
1) use strict equality (===) when comparing items
2) check for falsy/truthy values instead (`instructor.googleId`/`!instructor.googleId`)

Reply:
Should I change to
```
if (!instructor.googleId
            && (!instructorPrivilege && instructorPrivilege.privileges.courseLevel.canModifyInstructor)) {
```
?

Reply:
`instructor.googleId && instructorPrivilege && instructorPrivilege.privileges.courseLevel.canModifyInstructor` seems cleaner to me.